uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/leave-request-ctrl_test.js

```diff
@@ -1031,6 +1031,40 @@
           });
         });
 
+        describe('canSubmit', function () {
+          beforeEach(function () {
+            var status = optionGroupMock.specificValue('hrleaveandabsences_leave_request_status', 'value', '3');
+            var leaveRequest = LeaveRequestInstance.init(mockData.findBy('status_id', status));
+            leaveRequest.contact_id = CRM.vars.leaveAndAbsences.contactId.toString();
+            var directiveOptions = {
+              contactId: leaveRequest.contact_id, // staff's contact id
+              leaveRequest: leaveRequest,
+              userRole: 'staff'
+            };
+
+            initTestController(directiveOptions);
+          });
+
+          it('does not allow to submit the leave request without changes', function () {
+            expect($ctrl.canSubmit()).toBe(false);
+          });
+
+          describe('when a comment is added', function() {
+            beforeEach(function() {
+              $ctrl.request.comments.push({
```

Comment:
for simplicity's sake, i wonder if this would work even if you did 
```js
$ctrl.request.comments.push(jasmine.any(Object));
```
the structure of the comment object is not important for this test, isnt'it?